Add docker container for molecule testing

DrewHarvard opened the pull request:
Creates a container that will be used by Concourse to run Ansible and Molecule tests.

We require the version file to be prefixed temporarily until we make changes to the version calculator. More detail can be found in story `DVOP-1352`.

Resolves: DVOP-1418

DrewHarvard committed:
Add docker container for molecule testing

Creates a container that will be used by Concourse to run ansible and molecule tests.

Resolves: DVOP-1418

DrewHarvard committed:
Update version file

DrewHarvard committed:
Revert version file back to prefixed

We require the version file to be prefixed temporarily until we make changes to the version calculator.

We would have preferred just the version number so ECR is not duplicated in terms of the named tag however due to the current restrictions github will lose its context on the tags. Story raised in the backlog for work to be undertaken.

Jata2k commented on Dockerfile:
```diff
@@ -0,0 +1,33 @@
+FROM centos:8
+ENV container docker
```
Do we use this?

Jata2k commented on Dockerfile:
```diff
@@ -0,0 +1,33 @@
+                 molecule[docker] \
+                 molecule==${MOLECULE_VERSION}
+
+RUN rpm --import http://yum-repository.platform.aws.chdev.org/RPM-GPG-KEY-platform-noarch && \
```
I'd pull `yum-repository.platform.aws.chdev.org` into an `ARG`

Jata2k commented on README.md:
```diff
@@ -1,2 +1,26 @@
+
+The container has been built with the following versions installed on it:
+
+Name                    | Description                                                          | Example
```
I'd change this to just have columns: `Name` and `Version`, with values like `Ansible` and `2.9.10` respectively. Otherwise it's like we're describing versions rather than software as it were

DrewHarvard committed:
Address changes from pr

Removes unused ENV variable that was previously added.
Adds the yum repository url as an argument as suggested.
Update to the readme

Jata2k commented on Dockerfile:
```diff
@@ -0,0 +1,34 @@
+
+RUN rpm --import http://${YUM_REPOSITORY}/RPM-GPG-KEY-platform-noarch && \
+    yum install -y yum-utils && \
+    yum-config-manager --add-repo http://yum-repository.platform.aws.chdev.org/platform-noarch.repo && \
```
We need to make use of `YUM_REPOSITORY` here too

DrewHarvard committed:
Update yum repository url to use argument
